### ISSUE TITLE
Import Block with Provider Attribute targeting Resource in Module with Providers Block/Mapping Fails

### ISSUE THREAD

**Issue by BenPhegan** [hashicorp/terraform#35283]
### Terraform Version

```shell
Terraform v1.8.3
on linux_amd64
+ provider registry.terraform.io/databricks/databricks v1.20.0
+ provider registry.terraform.io/hashicorp/aws v5.7.0
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = "~> 1.3"
  required_providers {
    aws = "~> 5.0"
  }
}

# Default provider.  All these providers are pointing at the same account, but in a real-world scenario, you'd have different accounts using an `assume_role` block.
provider "aws" {
  region = "us-east-1"
}

# This is our first _alternate_ provider.  It's pointing at the same account, and in this account we already have a manually created bucket.
provider "aws" {
  region = "us-east-1"
  alias = "one"
}

# This is our second _alternate_ provider.  It's pointing at the same account, and in this account we don't have a manually created bucket.
provider "aws" {
  region = "us-east-1"
  alias = "two"
}

# This bucket exists, so we want to import it.  This could be any resource, but we're using an S3 bucket for simplicity.  Note we are using provider mappings to deploy to "different" accounts (just different providers in this example)
module "s3_bucket_one" {
    providers = { aws = aws.one }
    source = "./s3_bucket_module"
    bucket_name = "test-qwizzle-pop"
}

# This bucket does not, so we want to create it.  This could be any resource, but we're using an S3 bucket for simplicity.  Note we are setting the provider to deploy the same infrastructure in a different "account".
module "s3_bucket_two" {
    providers = { aws = aws.two }
    source = "./s3_bucket_module"
    bucket_name = "test-flibber-jibber"
}


# Here we are importing the bucket.  We have already identified the resource, as it is in a module.  We _expect_ to be able to identify the provider here.
# However, this does not work due to the code here: https://github.com/hashicorp/terraform/commit/5d6c5a9a3377ce316823ae3bb1fa8ecaf1ef1a60#diff-cd67749fe5d4685ebaeda3d69c0c4ca868ec6dace165b18811d6337f93aa7694R48
# To be clear, removing this appears to at least allow planning, but it's not clear if it will work when we apply.
# Essentially, it appears that the fact that providers in modules will _not_ match where there is a provider block in the module call has not been considered?
# This is peripherally discussed here: https://discuss.hashicorp.com/t/is-it-possible-with-import-block-to-a-module-to-use-a-provider-from-the-module/59232 but no reasonable workaround is provided.

import {
    to = module.s3_bucket_one.aws_s3_bucket.this
    id = "test-qwizzle-pop"
    provider = aws.one
}

# When attempting to "init", we get the following error:
#   ╷
#   │ Error: Invalid import provider argument
#   │ 
#   │   on main.tf line 48, in import:
#   │   48:     provider = aws.one
#   │ 
#   │ The provider argument can only be specified in import blocks that will generate configuration.
#   │ 
#   │ Use the providers argument within the module block to configure providers for all resources within a module, including imported resources.
#   ╵
#
# This is a problem, as we _need_ to be able to import resources from modules, and we _need_ to be able to specify the provider.
# What I expect is for the provider module mapping to be used to validate the provider, and then for the provider to be used to import the resource.
```

Module simply contains:
```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5"
    }
  }
}

resource "aws_s3_bucket" "this" {
  bucket = var.bucket_name
    acl    = "private"
}

variable "bucket_name" {
  type = string
  
}
```

### Debug Output

None provided.

### Expected Behavior

Ability to `init`, `plan` and apply with the result being a resource imported into module as per provider config in the `import` block.  Bonus points if there is validation to ensure that I am using the same provider in the `import` block as the `providers` block for the referenced module.

### Actual Behavior

It does not even correctly validate the configuration:

```
Initializing the backend...
Upgrading modules...
- s3_bucket_one in s3_bucket_module
- s3_bucket_two in s3_bucket_module
╷
│ Error: Invalid import provider argument
│ 
│   on main.tf line 48, in import:
│   48:     provider = aws.one
│ 
│ The provider argument can only be specified in import blocks that will generate configuration.
│ 
│ Use the providers argument within the module block to configure providers for all resources within a module, including imported resources.
╵
```

### Steps to Reproduce

1. `terraform init`

### Additional Context

Before I put a pull request together to remove or modify the config check, I want to ensure that my understanding of this bug is complete.  It appears that the actual generation of provider mappings has more context-aware import logic that correctly handles the issue above.  Just purely commenting out the check here: https://github.com/hashicorp/terraform/commit/5d6c5a9a3377ce316823ae3bb1fa8ecaf1ef1a60#diff-cd67749fe5d4685ebaeda3d69c0c4ca868ec6dace165b18811d6337f93aa7694R49 provides the ability to both init and plan, with the plan appearing to respect the passed, with no additional changes.

**Comment by jbardin** [comment]
Hi @BenPhegan,

Thanks for filing the issue. As described in the error message, the `provider` argument in the `import` block is only for config generation, however since you are importing into a module with existing configuration that option is not applicable. The import process should work correctly without specifying `provider` in the `import` block, since the provider is determined by the existing configuration. Are you having an issue importing without `provider` specified?

**Comment by BenPhegan** [comment]
The issues were originally related to a module that took two DataBricks providers as module mappings, I reproduced with the AWS provider as it is more universal (but did so with the expectation of a fix based on the issues I encountered with DataBricks).  Without adding explicit mappings on the DataBricks resources, Terraform appeared to be unable to find the resources and would fail stating that the resources to be imported could not be found.  Adding explicit mappings appeared to resolve the issue, as long as the config was accepted.

I can confirm that for the code above using the AWS provider, removing the provider statements does in fact allow the import to work correctly.

I will look at reproducing the original issue using only the DataBricks provider, and see if I can figure out the underlying issue.  It may not be a Terraform issue per se, but might relate to the DataBricks provider itself.
